server/src/core/tenant/datasource/services/datasource.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable, OnModuleDestroy, OnModuleInit } from '@nestjs/common';
+
+import { DataSource } from 'typeorm';
+import { PostgresConnectionOptions } from 'typeorm/driver/postgres/PostgresConnectionOptions';
+
+import { EnvironmentService } from 'src/integrations/environment/environment.service';
+import { CompanyEntity } from 'src/core/tenant/company-v2/company-v2.entity';
+
+@Injectable()
+export class DataSourceService implements OnModuleInit, OnModuleDestroy {
+  private mainDataSource: DataSource;
+  private connectionOptions: PostgresConnectionOptions;
+  private dataSources = new Map<string, DataSource>();
+
+  constructor(environmentService: EnvironmentService) {
+    this.connectionOptions = {
+      url: environmentService.getPGDatabaseUrl(),
+      type: 'postgres',
+      logging: false,
+      schema: 'public',
+    };
+    this.mainDataSource = new DataSource(this.connectionOptions);
+  }
+
+  private uuidToBase36(uuid: string) {
```

Comment:
workspace uuid are a bit too long for schema names, still not sure of this approach though